contracts/Aavegotchi/facets/ItemsRolesRegistryFacet.sol

```diff
@@ -303,6 +304,11 @@ contract ItemsRolesRegistryFacet is Modifiers, IERC7589, ERC1155Holder {
             _depositInfo.roleAssignment.expirationDate < block.timestamp || _depositInfo.roleAssignment.revocable,
             "ItemsRolesRegistryFacet: token has an active role"
         );
+
+        if(_depositInfo.roleAssignment.revocable || _depositInfo.roleAssignment.expirationDate != 0){
+            _unequipAllDelegatedWearables(_depositId, _depositInfo.deposit.tokenId);
+        }
```

Comment:
We don't need to check `revocable` and `expirationDate` because they are already checked on the `require` in line 303. However, we only need to unequip when it's a new grantee, meaning that the new user don't own the previously equipped Aavegotchis.
```suggestion
        if(_depositInfo.roleAssignment._grantee != _grantee) {
            // if depositId is being delegated to a new user, we need to make sure that Aavegotchis not owned by the new user are using these Wearables
            _unequipAllDelegatedWearables(_depositId, _depositInfo.deposit.tokenId);
        }
```